utils/manifest-utils/src/rules.rs

```diff
@@ -0,0 +1,146 @@
+use std::{
+    fs::OpenOptions,
+    io::BufReader,
+    path::{Path, PathBuf},
+    sync::{Arc, RwLock},
+};
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use structopt::StructOpt;
+
+#[derive(Clone, Debug, Default)]
+pub struct RuleStore {
+    pub path: PathBuf,
+    config: Arc<RwLock<RulesConfig>>,
+}
+
+impl RuleStore {
+    pub fn load_or_create(rules_file: &Path) -> Result<Self> {
+        if rules_file.exists() {
+            log::debug!("Loading rule from: {:?}", rules_file);
+            let file = OpenOptions::new().read(true).open(rules_file)?;
+
+            Ok(Self {
+                config: Arc::new(serde_json::from_reader(BufReader::new(file))?),
+                path: rules_file.to_path_buf(),
+            })
+        } else {
+            log::debug!("Creating default Rules configuration");
+            let config = Default::default();
+
+            let store = Self {
+                config: Arc::new(RwLock::new(config)),
+                path: rules_file.to_path_buf(),
+            };
+            store.save()?;
+
+            Ok(store)
+        }
+    }
+
+    fn save(&self) -> Result<()> {
+        log::debug!("Saving RuleStore to: {:?}", self.path);
+        Ok(std::fs::write(
+            &self.path,
+            serde_json::to_string_pretty(&*self.config.read().unwrap())?,
+        )?)
+    }
+
+    pub fn reload(&self) -> Result<()> {
+        log::debug!("Reloading RuleStore from: {:?}", self.path);
+        let new_rule_store = Self::load_or_create(&self.path)?;
+
+        self.replace(new_rule_store);
+
+        Ok(())
+    }
+
+    fn replace(&self, other: Self) {
+        let store = std::mem::take(&mut (*other.config.write().unwrap()));
+
+        *self.config.write().unwrap() = store;
+    }
+
+    pub fn set_everyone_mode(&self, mode: Mode) -> Result<()> {
+        log::debug!("Setting outbound everyone mode: {:?}", mode);
+        self.config.write().unwrap().outbound.everyone = mode;
+
+        self.save()
```

Comment:
If you acquire lock, you should wrap it in braces.
Otherwise there is danger, that saving will be done under lock
```suggestion
        {
                self.config.write().unwrap().outbound.everyone = mode;
        }
        self.save()
```

Reply:
I don't think so: in these cases when there is `self.save()`, previous LockGuard is dropped immediately (as it is not bound to longer lived value like in `utils/manifest_utils/src/policy.rs` line 224

Reply:
Yes, probably. But we had to many problems with locks that are held longer than we think and these problems are not easy to debug, so to be sure we always limit the scope of the lock explicite